nutanix/resource_nutanix_ndb_clone.go

```diff
@@ -0,0 +1,863 @@
+		res.DatabaseParameterProfileID = utils.StringPtr(databaseParameterProfileID.(string))
+	}
+	if snapshotID, ok := d.GetOk("snapshot_id"); ok {
+		res.SnapshotID = utils.StringPtr(snapshotID.(string))
+	}
+
+	if dbserverID, ok := d.GetOk("dbserver_id"); ok {
+		res.DbserverID = utils.StringPtr(dbserverID.(string))
+	}
+	if dbserverClusterID, ok := d.GetOk("dbserver_cluster_id"); ok {
+		res.DbserverClusterID = utils.StringPtr(dbserverClusterID.(string))
+	}
+	if dbserverLogicalClusterID, ok := d.GetOk("dbserver_logical_cluster_id"); ok {
+		res.DbserverLogicalClusterID = utils.StringPtr(dbserverLogicalClusterID.(string))
+	}
+	if createDbserver, ok := d.GetOk("create_dbserver"); ok {
+		res.CreateDbserver = createDbserver.(bool)
+	}
+	if clustered, ok := d.GetOk("clustered"); ok {
+		res.Clustered = clustered.(bool)
+	}
+	if nodeCount, ok := d.GetOk("node_count"); ok {
+		res.NodeCount = utils.IntPtr(nodeCount.(int))
+	}
+
+	if nodes, ok := d.GetOk("nodes"); ok {
+		res.Nodes = expandClonesNodes(nodes.([]interface{}))
+	}
+
+	if lcmConfig, ok := d.GetOk("lcm_config"); ok {
+		res.LcmConfig = expandLCMConfig(lcmConfig.([]interface{}))
+	}
+
+	if postgres, ok := d.GetOk("postgresql_info"); ok && len(postgres.([]interface{})) > 0 {
+		res.ActionArguments = expandPostgreSQLCloneActionArgs(d, postgres.([]interface{}))
+	}
+
+	if tags, ok := d.GetOk("tags"); ok && len(tags.([]interface{})) > 0 {
+		res.Tags = expandTags(tags.([]interface{}))
+	}
+	return nil
+}
+
+func expandClonesNodes(pr []interface{}) []*era.Nodes {
+	nodes := make([]*era.Nodes, len(pr))
+	if len(pr) > 0 {
+		for k, v := range pr {
+			val := v.(map[string]interface{})
+			node := &era.Nodes{}
+
+			if v1, ok1 := val["network_profile_id"]; ok1 && len(v1.(string)) > 0 {
+				node.Networkprofileid = utils.StringPtr(v1.(string))
+			}
+
+			if v1, ok1 := val["compute_profile_id"]; ok1 && len(v1.(string)) > 0 {
+				node.ComputeProfileID = utils.StringPtr(v1.(string))
+			}
+
+			if v1, ok1 := val["vm_name"]; ok1 && len(v1.(string)) > 0 {
+				node.Vmname = utils.StringPtr(v1.(string))
+			}
+
+			if v1, ok1 := val["nx_cluster_id"]; ok1 && len(v1.(string)) > 0 {
+				node.NxClusterID = utils.StringPtr(v1.(string))
+			}
+
+			if v1, ok1 := val["new_db_server_time_zone"]; ok1 && len(v1.(string)) > 0 {
+				node.NewDBServerTimeZone = utils.StringPtr(v1.(string))
+			}
+			if v1, ok1 := val["properties"]; ok1 && len(v1.([]interface{})) > 0 {
+				node.Properties = v1.([]*era.NodesProperties)
+			}
+
+			if v1, ok1 := val["dbserver_id"]; ok1 && len(v1.(string)) > 0 {
+				node.DatabaseServerID = utils.StringPtr(v1.(string))
+			}
+			nodes[k] = node
+		}
+		return nodes
+	}
+	return nil
+}
+
+func expandPostgreSQLCloneActionArgs(d *schema.ResourceData, pr []interface{}) []*era.Actionarguments {
+	if len(pr) > 0 {
+		args := []*era.Actionarguments{}
+
+		for _, v := range pr {
+			val := v.(map[string]interface{})
```

Comment:
we can optimize this by just iterating over map items and appending to list of specs as per key and value

Reply:
optimized.